sky/skylet/providers/azure/config.py

```diff
@@ -80,7 +82,27 @@ def _configure_resource_group(config):
     rg_create_or_update = get_azure_sdk_function(
         client=resource_client.resource_groups, function_name="create_or_update"
     )
-    rg_create_or_update(resource_group_name=resource_group, parameters=params)
+    rg_creation_start = time.time()
+    retry = 0
+    while time.time() - rg_creation_start < _WAIT_FOR_RESOURCE_GROUP_DELETION_TIMEOUT:
+        try:
+            rg_create_or_update(resource_group_name=resource_group, parameters=params)
+            break
+        except azure.exceptions().ResourceExistsError as e:
+            if "ResourceGroupBeingDeleted" in str(e):
+                if retry % 5 == 0:
+                    # TODO(zhwu): This should be shown in terminal for better
+                    # UX, which will be achieved after we move Azure to use
+                    # SkyPilot provisioner.
+                    logger.warning(
+                        f"Azure resource group {resource_group} is being deleted. "
+                        "It can only be provisioned after it is fully deleted. "
+                        "Waiting..."
```

Comment:
Should we add some info about why this happened? sth like `it is due to another cluster with the same cluster name being deleted`